### PR TITLE
ricardofalasca/django 2 compatibility

### PR DESCRIPTION
**Pull request by ricardofalasca**
fix(urls.reverse): update 'reverse' importing from django.urls instead urlresolvers

update(requirements): update required package versions of: django, pillow and sorl-thumbnail

fix(models): remove decorator @permalink due to deprecation, also return '' using mark_safe()

improvement(chardet): replaces chardet by cchardet due to better performance

fix(urls): change reverse importing due to urlresolvers was removed in Django 2.0, also add 'app_name' attribute on urls.py files

fix(migrations): 'on_delete' was missing on migrations

### CODE REVIEW

**Comment by lampslave** [comment]
Could you please fix tests?

File: setup.py [hunk 1]
```diff
@@ -6,16 +6,16 @@
...
     name='imagestore',
-    version='3.1.0',
+    version='3.2.0',
     packages=find_packages(),
     install_requires=[
-        'django',
-        'pillow',
-        'sorl-thumbnail',
+        'django>=1.11',
+        'pillow>=5.4.1',
```
Comment (lampslave) [inline setup.py]:
```suggestion
        'pillow',
```
I think we can use any version of pillow here.

**Comment by ricardofalasca** [comment]
> Could you please fix tests?

Fore sure. I'll do it.

**Comment by ricardofalasca** [comment]
@lampslave all fixed. Merge it when you have a chance please!

**Comment by lampslave** [comment]
Thank you!

**Pull request merged**